mono/mini/interp/interp.c

```diff
@@ -3881,8 +3881,24 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 				gpointer unboxed = mono_object_unbox_internal (this_arg);
 				sp [0].data.p = unboxed;
 			}
-retry_callvirt_fast:
-			if (imethod->code_type == IMETHOD_CODE_INTERP) {
+
+			InterpMethodCodeType code_type = imethod->code_type;
+
+			g_assert (G_UNLIKELY (code_type == IMETHOD_CODE_UNKNOWN) ||
+				code_type == IMETHOD_CODE_INTERP || code_type == IMETHOD_CODE_COMPILED);
```

Comment:
G_UNLIKELY is not needed here, g_assert already adds it.

Reply:
It is a little different, it is within the expression, not on the overall, inverted.
Compiler should kinda "back prop" from the later unlikely though that is outside the assert.
I'm willing to remove both asserts really.
And I suggest IMETHOD_CODE_UNKNOWN == 0, but that is separate.

Reply:
I removed the G_UNLIKELY.